File: Examples/Features/PDFExporting/ConvertExcelToPDF.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;

namespace GrapeCity.Documents.Excel.Examples.Features.PDFExporting
{
    public class ConvertExcelToPDF : ExampleBase
    {
        public override void Execute(GrapeCity.Documents.Excel.Workbook workbook)
        {
            Stream fileStream = this.GetResourceStream("xlsx\\Employee absence schedule.xlsx");
            workbook.Open(fileStream);
        }

        public override bool SavePdf
        {
            get
            {
                return true;
            }
        }

        public override bool ShowViewer
        {
            get
            {
                return false;
            }
        }
        public override string TemplateName
        {
            get
            {
                return "Employee absence schedule.xlsx";
            }
        }
        public override string[] UsedResources
        {
            get
            {
                return new string[] { "xlsx\\Employee absence schedule.xlsx" };
            }
        }
    }
}

File: Examples.VB/Features/PDFExporting/ConvertExcelToPDF.vb
Namespace Features.PDFExporting
    Public Class ConvertExcelToPDF
        Inherits ExampleBase
        Public Overrides Sub Execute(workbook As Excel.Workbook)
            'Open an excel file
            Dim fileStream = GetResourceStream("Employee absence schedule.xlsx")
            workbook.Open(fileStream)
        End Sub
        Public Overrides ReadOnly Property TemplateName As String
            Get
                Return "Employee absence schedule.xlsx"
            End Get
        End Property
        Public Overrides ReadOnly Property SavePdf As Boolean
            Get
                Return True
            End Get
        End Property
        Public Overrides ReadOnly Property ShowViewer As Boolean
            Get
                Return False
            End Get
        End Property
        Public Overrides ReadOnly Property IsNew As Boolean
            Get
                Return True
            End Get
        End Property
    End Class
End Namespace
